### PR TITLE
RPC code refactor

### PR DESCRIPTION
**Pull request by str4d**
Cherry-picked from the following upstream PRs:

- bitcoin/bitcoin#7307
- bitcoin/bitcoin#7348
- bitcoin/bitcoin#7766

### CODE REVIEW

**Comment by str4d** [comment]
@zkbot try

**Comment by str4d** [comment]
@mdr0id is seeing a couple of persistent RPC test failures locally; lets just double-check they aren't caused by something in master.

@zkbot retry

**Comment by bitcartel** [comment]
@zkbot try

**Comment by bitcartel** [comment]
@zkbot retry

**Comment by bitcartel** [comment]
ACK modulo a conflict in rpcmisc.cpp above. cc @str4d

**Comment by str4d** [comment]
Rebased on master to fix merge conflict.

@zkbot try

**Comment by str4d** [comment]
Rebased on master to fix merge conflicts and the bug @Eirik0 spotted.

**Comment by bitcartel** [comment]
@zkbot r+

File: src/wallet/rpcwallet.cpp [hunk 1]
```diff
@@ -4347,3 +4347,96 @@ UniValue z_listoperationids(const UniValue& params, bool fHelp)
...
+extern UniValue z_exportwallet(const UniValue& params, bool fHelp);
+extern UniValue z_importwallet(const UniValue& params, bool fHelp);
+
+extern UniValue z_getpaymentdisclosure(const UniValue& params, bool fHelp); // in rpcdisclosure.cpp
+extern UniValue z_validatepaymentdisclosure(const UniValue &params, bool fHelp);
+
+static const CRPCCommand commands[] =
+{ //  category              name                        actor (function)           okSafeMode
+    //  --------------------- ------------------------    -----------------------    ----------
+    { "rawtransactions",    "fundrawtransaction",       &fundrawtransaction,       false },
```
Comment (arcalinea) [inline src/wallet/rpcwallet.cpp]: Used to be wrapped by `#ifdef ENABLE_WALLET`

**Pull request merged**